docs/content/doc/developers/guidelines-backend.md

```diff
@@ -0,0 +1,111 @@
+---
+date: "2021-11-01T23:41:00+08:00"
+title: "Guidelines for Backend Development"
+slug: "guidelines-backend"
+weight: 20
+toc: false
+draft: false
+menu:
+  sidebar:
+    parent: "developers"
+    name: "Guidelines for Backend"
+    weight: 20
+    identifier: "guidelines-backend"
+---
+
+# Guidelines for Backend Development
+
+**Table of Contents**
+
+{{< toc >}}
+
+## Background
+
+Gitea uses Golang as the backend programming language. It uses many third-party packages and also write some itself. 
+For example, Gitea uses [Chi](https://github.com/go-chi/chi) as basic web framework. [Xorm](https://xorm.io) is an ORM framework that is used to interact with the database. 
+So it's very important to manage these packages. Please take the below guidelines before you start to write backend code.
+
+## Package Design Guideline
+
+### Packages List
+
+To maintain understandable code and avoid circular dependencies it is important to have a good code structure. The gitea backend is divided into the following parts:
```

Comment:
This is a general issue with the docs, but sometimes they use `Gitea` (e.g. line 24 in this file), and sometimes `gitea` (this line). It would be good to unify this spelling (probably `Gitea` is better).

Reply:
done.

Reply:
done.